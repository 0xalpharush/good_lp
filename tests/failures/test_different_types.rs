use good_lp::variables;

fn sametype<F>(_a: F, _b: F) {}

fn main() {
    let mut var1 = variables!();
    let mut var2 = variables!();
    sametype(var1, var2)
}