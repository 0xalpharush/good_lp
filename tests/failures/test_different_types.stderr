error[E0308]: mismatched types
 --> $DIR/test_different_types.rs:8:20
  |
6 |     let mut var1 = variables!();
  |                    ------------ the expected closure
7 |     let mut var2 = variables!();
  |                    ------------ the found closure
8 |     sametype(var1, var2)
  |                    ^^^^ expected closure, found a different closure
  |
  = note: expected struct `ProblemVariables<[closure@$DIR/src/variables_macro.rs:6:85: 6:89]>` (closure)
             found struct `ProblemVariables<[closure@$DIR/src/variables_macro.rs:6:85: 6:89]>` (closure)
  = note: no two closures, even if identical, have the same type
  = help: consider boxing your closure and/or using it as a trait object
